app/models/discussion/post.rb

```diff
@@ -0,0 +1,6 @@
+class Discussion::Post < ActiveRecord::Base
+  belongs_to :topic, class_name: Discussion::Topic.name, inverse_of: :posts
+  belongs_to :author, class_name: User.name, inverse_of: :posts
+  belongs_to :parent, class_name: Discussion::Post.name, inverse_of: :children
+  has_many :children, class_name: Discussion::Post.name, inverse_of: :parent, dependent: :destroy
```

Comment:
Consider using gem `edge` to handle the parent/children relation ?  (same as materials)